src/Simplex/Chat/Store/Messages.hs

```diff
@@ -1083,37 +1084,63 @@ getDirectChatAround_ db user ct@Contact {contactId} aroundItemId count search =
   let chatItems = beforeChatItems <> [middleChatItem] <> afterChatItems
   pure $ Chat (DirectChat ct) chatItems stats
 
-getGroupChat :: DB.Connection -> VersionRangeChat -> User -> Int64 -> ChatPagination -> Maybe String -> ExceptT StoreError IO (Chat 'CTGroup)
+getDirectChatInitial_ :: DB.Connection -> User -> Contact -> Int -> ExceptT StoreError IO (Chat 'CTDirect, ChatLandingSection)
+getDirectChatInitial_ db user@User {userId} ct@Contact {contactId} count = do
+  firstUnreadItem <- liftIO getDirectChatMinUnreadItemId_
+  case firstUnreadItem of
+    Just firstUnreadItemId -> do
+      chat <- getDirectChatAround_ db user ct firstUnreadItemId count ""
+      let Chat {chatItems} = chat
+      let chatItemIds = map (cchatItemId) chatItems
+      lastItemId <- liftIO $ getDirectChatItemIdsLast_ db user ct 1 ""
+      let lastItemIdInChat = head lastItemId `elem` chatItemIds
```

Comment:
safer to pattern match maybe.. though we're already in branch where we know there's at least 1 item.

also probably more efficient to use foldr for single pass over list (instead of map, elem) to find if last item id is in list

Reply:
(we expect it last item id to match end of list)